File: src/models/UserRole.ts
export type UserRole = 'customer' | 'admin' | 'driver';

export const USER_ROLES: UserRole[] = ['customer', 'admin', 'driver'];

export const userRoleDisplayName: Record<UserRole, string> = {
  customer: 'Customer',
  admin: 'Admin',
  driver: 'Driver',
};

export const userRoleIconName: Record<UserRole, string> = {
  customer: 'person', // you can map to an icon library name
  admin: 'shield',
  driver: 'car',
};

File: SmartFoodDeliveryApp/Models/UserRole.swift
import Foundation

nonisolated enum UserRole: String, Codable, Sendable, CaseIterable, Identifiable {
    case customer
    case admin
    case driver

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .customer: "Customer"
        case .admin: "Admin"
        case .driver: "Driver"
        }
    }

    var icon: String {
        switch self {
        case .customer: "person.fill"
        case .admin: "shield.fill"
        case .driver: "car.fill"
        }
    }
}
